packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -921,7 +921,9 @@ export class AuthTransformer extends TransformerAuthBase implements TransformerA
       const roleDefinition = this.roleMap.get(role)!;
       roleDefinition.areAllFieldsAllowed = allowedFields.length === fields.length;
       roleDefinition.areAllFieldsNullAllowed = nullAllowedFields.length === fields.length;
-      roleDefinition.allowedFields = this.addAutoGeneratedFields(ctx, def, allowedFields, fields);
+      // include primary key in allowed fields on update
+      // the primary key won't be updated, but this indicates that including the primary key in the input should not result in unauth err
+      roleDefinition.allowedFields = this.addAutoGeneratedFields(ctx, def, [...allowedFields, getObjectPrimaryKey(def).name.value], fields);
```

Comment:
When customer explicitly define a primary key, it may be added twice to the array `[...allowedFields, getObjectPrimaryKey(def).name.value]`. Does it impact any functionality?

Reply:
`addAutoGeneratedFields` first makes a set out of the supplied `allowedFields`. I can add a test to ensure this behavior.